### PR TITLE
Document fix about AI item pointer overflowing

### LINKED ISSUE #1065: Bug: After a trainer uses its 1st item, it'll start reading past the item list for items to use
https://twitter.com/pimanrules/status/1528137472680640513

https://youtu.be/Q6E6OaWb7LQ?t=1002

### PR DESCRIPTION
**Pull request by Idain**
Fixes #1065

### CODE REVIEW

**Review by vulcandth — CHANGES_REQUESTED** [review]

**Review by vulcandth — APPROVED** [review]
Looks good. Idain verified the bug exists and the bugfix works.

File: docs/bugs_and_glitches.md [hunk 1]
```diff
@@ -1455,6 +1456,33 @@ Pryce's dialog ("That BADGE will raise the SPECIAL stats of POKéMON.") implies
  	ret
 ```
 
+### AI might use its base reward's value as an item
+
+In the `AI_TryItem` routine, the code checks both `wEnemyTrainerItem1` and `wEnemyTrainerItem2` to see if the AI's items match any in the `AI_Items` list. However, if the AI has used its first item (or it's set to `ITEM_NONE`) and doesn't use its second item, the item pointer can overflow to the subsequent address, `wEnemyTrainerBaseReward`. If this address's value matches an item in the `AI_Items` list, the AI could mistakenly use it.
```
Comment (Rangi42) [inline docs/bugs_and_glitches.md:1461]:
```suggestion
In the `AI_TryItem` routine, an item pointer is set to `wEnemyTrainerItem1` and then increments to `wEnemyTrainerItem2` to see if either of the AI's items are in the `AI_Items` list. However, if the AI has used its first item (or its first one is `ITEM_NONE`) and hasn't used its second item, the item pointer can increment from `wEnemyTrainerItem2` to `wEnemyTrainerBaseReward`. If the value at this address then matches an item in the `AI_Items` list, the AI could mistakenly use it.
```

File: engine/battle/ai/items.asm [hunk 2]
```diff
@@ -175,6 +175,7 @@ AI_TryItem:
 	ld b, h
 	ld c, l
 	ld hl, AI_Items
+; BUG: AI might use its base reward's value as an item (see docs/bugs_and_glitches.md)
```
Comment (Rangi42) [inline engine/battle/ai/items.asm:178]:
```suggestion
; BUG: AI might use its base reward value as an item (see docs/bugs_and_glitches.md)
```

File: docs/bugs_and_glitches.md [hunk 3]
```diff
@@ -1455,6 +1456,33 @@ Pryce's dialog ("That BADGE will raise the SPECIAL stats of POKéMON.") implies
...
+  	...
+ 	ld a, [wTrainerClass]
+ 	dec a
+ 	ld hl, TrainerClassAttributes + TRNATTR_AI_ITEM_SWITCH
+ 	ld bc, NUM_TRAINER_ATTRIBUTES
+ 	call AddNTimes
+ 	ld b, h
+ 	ld c, l
+ 	ld hl, AI_Items
+-; BUG: AI might use its base reward's value as an item (see docs/bugs_and_glitches.md)
```
Comment (Rangi42) [inline docs/bugs_and_glitches.md:1476]:
```suggestion
-; BUG: AI might use its base reward value as an item (see docs/bugs_and_glitches.md)
```

File: docs/bugs_and_glitches.md [hunk 4]
```diff
@@ -1455,6 +1456,33 @@ Pryce's dialog ("That BADGE will raise the SPECIAL stats of POKéMON.") implies
  	ret
 ```
 
+### AI might use its base reward's value as an item
```
Comment (Rangi42) [inline docs/bugs_and_glitches.md:1459]:
```suggestion
### AI might use its base reward value as an item
```

File: docs/bugs_and_glitches.md [hunk 5]
```diff
@@ -59,6 +59,7 @@ Fixes in the [multi-player battle engine](#multi-player-battle-engine) category
   - [AI makes a false assumption about `CheckTypeMatchup`](#ai-makes-a-false-assumption-about-checktypematchup)
   - [AI use of Full Heal or Full Restore does not cure Nightmare status](#ai-use-of-full-heal-or-full-restore-does-not-cure-nightmare-status)
   - [AI use of Full Heal does not cure confusion status](#ai-use-of-full-heal-does-not-cure-confusion-status)
+  - [AI might use its base reward's value as an item](#ai-might-use-its-base-rewards-value-as-an-item)
```
Comment (Rangi42) [inline docs/bugs_and_glitches.md:62]:
```suggestion
  - [AI might use its base reward value as an item](#ai-might-use-its-base-reward-value-as-an-item)
```

**Review by Rangi42 — APPROVED** [review]

**Pull request merged**